react-design/my-app/src/RotatingBanner.jsx

```diff
@@ -0,0 +1,31 @@
+import  './styles.css'
+import Banner from './Banner';
+import PrevButton from './PrevButton';
+import NextButton from './NextButton';
+import Indicators from './Indicators';
+import { useState } from 'react';
+
+export default function RotatingBanner ({items}) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  function handleClickPrev(e) {
+    setCurrentIndex((currentIndex - 1 + items.length) % items.length);
+  }
+
+  function handleClickNext(e) {
+    setCurrentIndex((currentIndex + 1) % items.length);
+  }
+
+  function handleClickIndicators(e) {
+    setCurrentIndex(+e.target.textContent)
+  }
```

Comment:
I'd generally suggest against passing whole event objects up the component chain. Instead, write your callback props to expect the values they need (in this case, just the index to switch to).